TicTacToe2/TicTacToe/Board.cs

```diff
@@ -7,32 +7,27 @@ namespace TicTacToe2.TicTacToe
     class Board
     {
         public int Size { get; }
-        public char[,] Spaces;
+        private readonly char[,] spaces;
 
         public Board(int size)
         {
             Size = size;
-            Spaces = new char[Size, Size];
+            spaces = new char[Size, Size];
         }
 
         public void PlaceMark(int x, int y, char symbol)
         {
-            Spaces[x, y] = symbol;
+            spaces[x, y] = symbol;
         }
 
         public bool HasMarkAt(int x, int y)
         {
-             if (Spaces[x, y] == 'X' || Spaces[x, y] == 'O')
-            {
-                return true;
-            }
-            return false;
-
+            return spaces[x, y] != ' ';
         }
 
         public bool IsOnBoard(int x, int y)
         {
-            if (x <= Size - 1 && y <= Size - 1)
+            if (x >= Size - 1 || x < 0 || y >= Size - 1 || y < 0)
```

Comment:
This logic doesn't work out. We want to be sure that the point given is on the board. "On the board" can be defined by these rules

- X is greater than or equal to 0
- X is less than the size of the board
- Y is greater than or equal to 0
- Y is less than the size of the board

All of these conditions need to be true for the point to be considered to be on the board. Typically I would write the body of this function as

```c#
return x >= 0 && x < Size && y >= 0 && y < Size;
```

Your logic is pretty close to the inverse of this. If you put return false in the if statement the function would be pretty close to logically correct. The if statement condition would need to be changed slightly to

```c#
x >= Size || x < 0 || y >= Size || y < 0
```

as a point with x or y = Size - 1 should be considered to be on the board.